cmd/skopeo/proxy.go

```diff
@@ -266,6 +268,23 @@ func (h *proxyHandler) openImageImpl(args []any, allowNotFound bool) (replyBuf,
 		return ret, err
 	}
 
+	unparsedTopLevel := image.UnparsedInstance(imgsrc, nil)
+	policy, err := signature.DefaultPolicy(h.sysctx)
+	if err != nil {
+		return ret, err
+	}
+	policyContext, err := signature.NewPolicyContext(policy)
+	if err != nil {
+		return ret, err
+	}
+	allowed, err := policyContext.IsRunningImageAllowed(context.Background(), unparsedTopLevel)
```

Comment:
Here'd what we want is a new API like `IsRunningImageSigned` and it would fail if the signature policy is `insecureAcceptAnything` for https://github.com/ostreedev/ostree-rs-ext/issues/79